spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/RetryLoadBalancerInterceptor.java

```diff
@@ -159,15 +175,24 @@ private RetryTemplate createRetryTemplate(String serviceName, HttpRequest reques
 		if (retryListeners != null && retryListeners.length != 0) {
 			template.setListeners(retryListeners);
 		}
-		template.setRetryPolicy(!properties.getRetry().isEnabled() || retryPolicy == null ? new NeverRetryPolicy()
+		template.setRetryPolicy(!getLoadBalancerProperties(serviceName).getRetry().isEnabled() || retryPolicy == null ? new NeverRetryPolicy()
 				: new InterceptorRetryPolicy(request, retryPolicy, loadBalancer, serviceName));
 		return template;
 	}
 
 	private String getHint(String serviceId) {
-		String defaultHint = properties.getHint().getOrDefault("default", "default");
-		String hintPropertyValue = properties.getHint().get(serviceId);
+		String defaultHint = getLoadBalancerProperties(serviceId).getHint().getOrDefault("default", "default");
+		String hintPropertyValue = getLoadBalancerProperties(serviceId).getHint().get(serviceId);
 		return hintPropertyValue != null ? hintPropertyValue : defaultHint;
 	}
 
+	@Deprecated
+	private LoadBalancerProperties getLoadBalancerProperties(String serviceId) {
+		if (propertiesFactory != null) {
+			return propertiesFactory.getLoadBalancerProperties(serviceId);
```

Comment:
I think this isn't quite right. If there isn't properties for a service, the properties object should be returned.

Reply:
Factory method returns `default` properties when service-specific properties were not provided.
```
private LoadBalancerProperties getLoadBalancerServiceProperties(String serviceName) {
	return servicesProperties.getClient().getOrDefault(serviceName, servicesProperties.getDefaultClientProperties());
}
```